packages/remote-core/src/serialization/FlowThreadSerialization.ts

```diff
@@ -24,10 +25,13 @@ export class FlowThreadSerialization extends ThreadSerializationStructuredClone
           for (const serializer of serializers) {
             const serialization = serializer.serialize(val);
             if (serialization.applied) {
-              return serialization.result;
+              return serialize(serialization.result);
             }
           }
           if (isObjectType(val)) {
+            if ((val as never)[TRANSFERABLE]) {
+              return serialize(val);
+            }
```

Comment:
I would prefer to see this check in the `isSerializableByBase` function, because its actually a "just use the base serialization"-case. And maybe use `TRANSFERABLE in val` instead of `(val as never)[TRANSFERABLE]`.

```typescript
  private isSerializableByBase(val: unknown) {
    return (
      val instanceof Map ||
      val instanceof Set ||
      Array.isArray(val) ||
      typeof val === "function" ||
      (isObjectType(val) && TRANSFERABLE in val)
    );
  }
   ```